### ISSUE TITLE
Update to Font Awesome v5

### ISSUE THREAD

**Issue by MichaelBelgium** [flarum/core#1326]
A suggestion by [Eldenroot](https://discuss.flarum.org/u/Eldenroot) from [this forum topic](https://discuss.flarum.org/d/8018-update-to-fontawesome-icon-v5)

I must say also those icons look very neat! I like that u can have 3 types of the same icon (solid, regular, light)

[Font awesome v5](https://fontawesome.com/)

**Comment by Eldenroot** [comment]
Yes, these new icons are just beautiful. I hope it will be included in v0.1 release

**Comment by franzliedke** [comment]
@tobscure What do you think?

In any case, version 5 does not seem to be available yet in the [shim repository we use for including font-awesome via Composer](https://github.com/components/font-awesome).

**Comment by tobscure** [comment]
I love the new icons, but I have a couple of issues with this:

1. Font Awesome 5 does not appear to be open-source / on [GitHub](https://github.com/FortAwesome/Font-Awesome) - hence the Composer repo hasn't been updated. We can't really ship it without being able to pull it in with Composer.

2. Their SVG JS...

> Our dash of performant JS replaces your fas fa-icon and CSS :after icon elements with SVG.

*Dash* of *performant* JS?! It's *271 KB* gzipped. ***642 KB*** uncompressed. That's bigger than all of Flarum's JS.

Not that we'd have to use the SVG JS. We could stick with the icon font. But still, such an outrageous claim 😆

**Comment by MichaelBelgium** [comment]
The open source thing still needs to happen i think, as well as [the changelog](https://github.com/FortAwesome/Font-Awesome/issues/12035)

**Comment by luceos** [comment]
Perhaps we should simply consider allowing icons to be interchangeable? That would allow an extension to switch to FA5 or material icons.

**Comment by tobscure** [comment]
@luceos how would you propose to achieve that?

**Comment by luceos** [comment]
I haven't considered implementation yet, although I should have. Perhaps by making the icons an extension by itself and only adding references to the pages? Then in the frontend we read a icon map and interchange the icons with their corresponding mapping. Possibly we could (preferably should) achieve this during compilation.

Is this a valid suggestion? Your frontend foo is stronger than mine @tobscure . 😊

**Comment by MichaelBelgium** [comment]
@tobscure Just noticed FontAwesome 5.0.6 has been [released](https://github.com/FortAwesome/Font-Awesome/releases/tag/5.0.6) some days ago

**Comment by franzliedke** [comment]
We need the update to reach [this repository](https://github.com/components/font-awesome), though...

**Comment by MichaelBelgium** [comment]
I feel like they're waiting for a pull request for that 🤔 Previous updates has been done via that, most of the time if i see correctly.

**Comment by franzliedke** [comment]
And [there it is](https://github.com/components/font-awesome/pull/47). :grinning: 

If anybody is interested in sending a PR that updates Flarum to the new version, go for it. :+1:

**Comment by tobscure** [comment]
And please include file-size comparisons :D

**Comment by afrastgeek** [comment]
Hi, everybody!

I've been trying out to implement this for some times, and just now created a PR for this (even though still WIP).

Any comments are welcome, and please let me know if i'm doing something wrong 😀

**Comment by franzliedke** [comment]
Fixed by @afrastgeek in #1372 and all the companion PRs. :ok_hand:

**Comment by MichaelBelgium** [comment]
Lovely, great work @afrastgeek !

**Comment by luceos** [comment]
I'm reopening this because the PR we merged (in all extensions as well) use the `fa` prefix, which is an invalid prefix. Valid prefixes are:

- `far` for regular
- `fab` for brands
- `fas` for strong
- `fal` for light

We decided to remove the shim (fallback). And:

![image](https://user-images.githubusercontent.com/504687/39758757-1a5e4200-52d1-11e8-9ff2-51c88f9adef4.png)

**Comment by afrastgeek** [comment]
@luceos thanks for the heads up. I wasn't notice the deprecation on `fa` prefix back then.

anyway from the [Font Awesome 5 Upgrade Guide](https://fontawesome.com/how-to-use/upgrading-from-4), the new icon prefix are not all `fas`, some has `far` prefix. The `far` prefix are the new syntax for the old `icon-name-o` open icon.

In some component, like the follow button or the unread thread status, they're using both solid and open (now called regular) icon for toggle. So, I suggest to not replace the `far` icon prefix.

Let's just change the `fa` prefix to `fas`.

**Comment by luceos** [comment]
@afrastgeek that's a good point, let's stick to those then. 

I also noticed the deprecated `fa` still works, but it would be better to change them to prevent confusion and possible breaks in the future.

**Comment by luceos** [comment]
@CDK2020 thanks 🤗  I've merged your PR to core, now we just need to do all core extensions.

**Comment by datitisev** [comment]
Just in case, I'd recommend looking at all the places where icons are used, as the names of some were changed, resulting in breaking changes.

**Comment by franzliedke** [comment]
If I remember correctly, that was done as part of the original set of PRs related to this issue.

**Comment by sijad** [comment]
some icons missing, at least in Suspend and Subscribe extensions.

![image](https://user-images.githubusercontent.com/7693001/41106131-a1eb5ca6-6a84-11e8-977a-211ddc8f94ca.png)
![image](https://user-images.githubusercontent.com/7693001/41106149-ad8abcdc-6a84-11e8-9444-a2db12b6094b.png)
![image](https://user-images.githubusercontent.com/7693001/41106153-b084d0bc-6a84-11e8-829f-b241b444de77.png)

**Comment by luceos** [comment]
@sijad did you recompile the js for those extensions?

**Comment by sijad** [comment]
no I didn't, I've cloned `flarum/flarum`, changed all dependencies version to `dev-master`, then `composer install`, installed Flarum. and some icons don't show up.

**Comment by sijad** [comment]
is it something that should be fixed somehow? I think users shouldn't have to recompile assets after installation/upgrade.

**Comment by luceos** [comment]
We decided not to commit the dist files along every single time. In the future there will be a bot (or ci) that does the recompilation for us. The reason for this is that it causes too many issues with merging and resolving conflicts. In addition it just increases the size of a diff.

That you have to recompile is because you're using a non-tagged version. Tagged versions will absolutely have properly compiled dist files.

**Comment by PartialDevAdmin** [comment]
Is this for beta8 only? What version of FontAwesome is beta7.1 running?

**Comment by TurboProgramming** [comment]
> What version of FontAwesome is beta7.1 running?

4.6

> Is this for beta8 only

Yes

Hope these answers are correct 😬
